waltz-test/src/main/python/waltz_ducktape/tests/validation/connection_interruption_test.py

```diff
@@ -77,15 +94,96 @@ def client_server_network_interruption(self, validation_cmd, timeout, interrupt_
                     'Network interruption failed, newly stored transactions detected'
 
-                # enable connection on port
-                self.enable_traffic_to_port(node, self.waltz_server.port)
             finally:
                 """
                 delete the added iptable rule as it is not removed with the end of waltz process and could persist on 
-                waltz-server VM, if process is signaled to end (^C) or an exception is thrown during execution of the try block. 
-                If iptable rule is not present (already deleted) nothing happens on the waltz-server VM side.
+                waltz-server VM, if process is signaled to end (^C) or an exception is thrown during execution of the try block.
                 """
                 self.enable_traffic_to_port(node, self.waltz_server.port)
 
         wait_until(lambda: self.verifiable_client.task_complete() == True, timeout_sec=timeout,
                    err_msg="verifiable_client failed to complete task in %d seconds." % timeout)
+
+    class StorageNodeInfo:
+        """
+        Representation of a storage node and number of transactions stored on a node
+        """
+        def __init__(self, node):
+            self.node = node
+            self.total_number_of_existing_transactions_across_partitions = 0
+
+    def storage_node_network_interruption(self, validation_cmd, num_active_partitions, txn_per_client, num_clients, timeout,
+                                          interrupt_duration, num_of_nodes_to_bounce):
+        """
+        A validate function to test bouncing network connection between server and storage. Verification of correctness
+        is done by comparing expected number of stored transactions with current transactions in the Waltz Cluster.
+
+        :param validation_cmd: The command that is send to ClientCli
+        :param num_active_partitions: Number of active partitions
+        :param txn_per_client: Number of transactions per client
+        :param num_clients: Number of waltz clients
+        :param timeout: Test timeout
+        :param interrupt_duration: Duration (milliseconds) of communication interruption between server and storage
+        :param num_of_nodes_to_bounce: Number of storage nodes to bounce. This may affect the quorum
+        """
+
+        bounced_nodes = []
+        for node_number in sample(range(len(self.waltz_storage.nodes)), num_of_nodes_to_bounce):
+            bounced_nodes.append(self.StorageNodeInfo(self.waltz_storage.nodes[node_number]))
+
+        admin_port = self.waltz_storage.admin_port
+        port = self.waltz_storage.port
+
+        # Step 1: Get sum of current max_transaction_ids
+        for bounced_node_info in bounced_nodes:
+            for partition in range(num_active_partitions):
+                bounced_node_info.total_number_of_existing_transactions_across_partitions += max(-1, self.get_storage_max_transaction_id(
```

Comment:
why are we setting the `total_number_of_existing_transactions_across_partitions` to negative value if there are no transaction written to the Waltz Cluster?
If we set the num_active_partitions to 2 then total_number_of_existing_transactions_across_partitions is set to -2.
Can we fix this to set the variable to 0 to reflect the actual number of transactions written to Waltz Cluster.